Adding maturity field

andrewyatz opened the pull request:
Issue #33 raised a need for a maturity field. This adds that field in, types it to an enum and makes it a required field. The enum requirement could be removed at a later date.

andrewyatz committed:
Adding maturity field

Issue #33 raised a need for a maturity field. This adds that field in, types it to an enum and makes it a required field. The enum requirement could be removed at a later date.

mcupak reviewed (changes requested):
Can we make it not required? I'd bet most people won't need it.

jrambla commented:
The ones requiring it are the *clients* of the service, not the owners. It must be required if it should be of any utility, on the contrary we could end up in the same situation that we are today. Forcing to choose an option would make the owners to think about it.

andrewyatz commented:
I agree with @jrambla that it should be required and why I added `unknown` in to give those implementing it a way to shortcut the response. If we were to remove `unknown` then I push for it to be optional

andrewyatz committed:
Removing maturity as a required field 

No longer have the unknown state and remove the field as required. Effectively having unknown was equivalent to making the field optional.

andrewyatz committed:
Adding in a valid example field and renaming to environment

We decided maturity was not communicating the right ideas behind the field and instead switched to environment e.g. "the production environment"

mcupak commented:
Agreed to rename to `environment` and replace `unknown` with `null` (i.e. make optional).

mcupak reviewed (approved):
LGTM now
